api/cpp/esp-idf/slint/README.md

```diff
@@ -11,81 +11,110 @@ It has been tested on ESP32-S3 devices.
 ```
-
-this means you are not using the right toolchain.
-
-
-## Usage
-
-By using this component, the `Slint::Slint` CMake target is linked to your application and you can access the entire functionality of the
-[Slint C++ API](https://slint.dev/docs/cpp).
-
-In addition, this component provides the `slint-esp.h` header file, which provides a a function to initialize the Slint backend
-that must be called before setting up the UI.
-Behind the scene, it will use `esp_lcd_panel_draw_bitmap` to render to the screen,
-and is base on [ESP LCD Touch](https://components.espressif.com/components/espressif/esp_lcd_touch) for the touch events.
-
-Use this platform implementation by instantiating it with a `esp_lcd_panel_handle_t`, an optional `esp_lcd_touch_handle_t`, and a pointer to one
-or two `slint::platform::Rgb565Pixel` frame buffers. Next, register the instance with the Slint run-time library by calling `slint_esp_init()`:
-
-```cpp
-#include "slint-esp.h"
-// ...
-
-
-// Initialize the panel and touch handle
-esp_lcd_panel_handle_t panel_handle = NULL;
-bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
-bsp_display_backlight_on();
-esp_lcd_touch_handle_t touch_handle = NULL;
-bsp_touch_new(&bsp_touch_cfg, &touch_handle);
-
-// Allocate a frame buffer
-static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
-
-// initialize the Slint ESP backend
-slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
-                                   touch_handle, buffer);
-
+import { VerticalBox, AboutSlint } from "std-widgets.slint";
+export component AppWindow inherits Window {
+    VerticalBox {
+        AboutSlint {}
+        Text {
+            text: "Hello World";
+            font-size: 18px;
+            horizontal-alignment: center;
+        }
+    }
+}
 ```
-
-Alternatively, you can implement your own sub-class of `slint::platform::Platform` to drive the screen and handle input events.
-
-Next, integrate your `.slint` files into the build by compiling them to C++ and linking them to your component:
-
+1. Edit `main/CMakeLists.txt` to adjust for the new `slint-hello-world.cpp`, add `slint` as required component,
+   and instruction the build system to compile `appwindow.slint` to `appwindow.h`. The file should look like this:
 ```cmake
-slint_target_sources(${COMPONENT_LIB} my_application_ui.slint)
-```
-
-Instantiate the Slint component by including the generated header file (`my_application_ui.h` in the above example), call `create()`
-on the generated class and `run()` to spin the event loop:
-
-```cpp
-#include "my_application_ui.h"
-
-/// ...
-
-auto my_application_ui = MainWindow::create();
-
-my_application_ui->run();
+idf_component_register(SRCS "slint-hello-world.cpp" INCLUDE_DIRS "." REQUIRES slint)
+slint_target_sources(${COMPONENT_LIB} appwindow.slint)
 ```
+1. Open the configuration editor with `idf.py menuconfig`:
```

Comment:
This step fails for me:

```
-- Building ESP-IDF components for target esp32
Dependencies lock doesn't exist, solving dependencies.
CMake Error at /home/olivier/esp/esp-idf/tools/cmake/build.cmake:540 (message):
  HINT: Component "espressif/esp-box" has suitable versions for other
  targets: "esp32s3".  Is your current target "esp32" set correctly?

  ERROR: Solver failed processing dependency "espressif/esp-box" from the
  manifest file "/home/olivier/esp/slint-hello-world/main/idf_component.yml".

  Cannot find versions of "espressif/esp-box" satisfying "*" for the current
  target "esp32".

Call Stack (most recent call first):
  /home/olivier/esp/esp-idf/tools/cmake/project.cmake:464 (idf_build_process)
  CMakeLists.txt:8 (project)
```


`idf.py set-target esp32s3`  fixed it almost, then i get 

```
Dependencies lock doesn't exist, solving dependencies.
CMake Error at /home/olivier/esp/esp-idf/tools/cmake/build.cmake:540 (message):
  HINT: Component "slint/slint" has some pre-release versions:
  "1.2.0-alpha4", "1.2.0-alpha3" satisfies your requirements.  To allow
  pre-release versions add "pre_release: true" to the dependency in the
  manifest.

  ERROR: Solver failed processing dependency "slint/slint" from the manifest
  file "/home/olivier/esp/slint-hello-world/main/idf_component.yml".

  Cannot find versions of "slint/slint" satisfying "*" for the current target
  "esp32s3".

Call Stack (most recent call first):
  /home/olivier/esp/esp-idf/tools/cmake/project.cmake:464 (idf_build_process)
  CMakeLists.txt:8 (project)
```

So i edited my idf_component.yml to set the exact version and then i could get it work.

Reply:
Yes, either that or `pre_release: true` as mentioned in https://github.com/slint-ui/slint/pull/3324#issuecomment-1688416532 - so I think that issue will go away with the release.